### DISCUSSION TITLE
[Enhancement] [tree] 增加节点双击事件（Add node double-click event）

### DISCUSSION THREAD

**Discussion by WangJincheng4869** [element-plus/element-plus#16205] [Feature Request]
Feature type: **`Enhancement`**

## Details

- Enhanced component name: `tree`

## Description

1. tree 节点的双击事件还是很常用的，比如我关闭了点击展开节点，可以是用双击事件展开节点。
2. 如果当菜单使用，比如可以用点击事件展开节点，双击事件就可以使选中节点。（反之也可以）

> 1. The double click event of a tree node is still very common, for example, if I close the click to expand the node, I can use the double click event to expand the node.
> 2. If the menu is used, for example, clicking events can expand nodes, and double clicking events can select nodes. (Conversely, it can also be done)

### Additional comments

_(empty)_

**Comment by L0st1** [comment]
这个可以采用[自定义节点](https://element-plus.gitee.io/zh-CN/component/tree.html#%E8%87%AA%E5%AE%9A%E4%B9%89%E8%8A%82%E7%82%B9%E5%86%85%E5%AE%B9)实现，应该是不用暴露双击事件（感觉很少不使用自定义节点）

**Reply by WangJincheng4869** [reply]
得考虑一下二封啊，自定义节点的插槽还是要暴露出去的，有这个事件二封能更方便，并且用起来有也方便，只想要个双击事件，没必要写个插槽搞自定义节点啊。

**Comment by WangJincheng4869** [comment]
1. 其次个人理解组件库的存在就是为了方便开发，简化各种复杂的操作。如果意味的只是甩出一个插槽直接自定义那也就失去了组件库的易用性。
2. 另外 tree 组件完全可以参考老前辈 [ztree](https://github.com/zTree/zTree_v3)，这个 tree 插件讲过长达十多年的积累，可以说是前端最好用的 tree 组件库，没有之一。唯一的遗憾就是依赖 jQuery。
